src/mame/drivers/vii.cpp

```diff
@@ -609,90 +635,123 @@ void vii_state::switch_bank(uint32_t bank)
 	}
 }
 
-void vii_state::do_gpio(uint32_t offset)
+uint16_t spg2xx_cart_state::do_spg243_vii_io(uint16_t what, int index)
 {
-	uint32_t index  = (offset - 1) / 5;
-	uint16_t buffer = m_io_regs[5*index + 2];
-	uint16_t dir    = m_io_regs[5*index + 3];
-	uint16_t attr   = m_io_regs[5*index + 4];
-	uint16_t special= m_io_regs[5*index + 5];
+	if(index == 1)
+	{
+		uint32_t bank = ((what & 0x80) >> 7) | ((what & 0x20) >> 4);
+		switch_bank(bank);
+	}
+	return what;
+}
 
-	uint16_t push   = dir;
-	uint16_t pull   = (~dir) & (~attr);
-	uint16_t what   = (buffer & (push | pull));
-	what ^= (dir & ~attr);
-	what &= ~special;
+uint16_t spg2xx_cart_state::do_spg243_vsmile_io(uint16_t what, int index)
+{
+	// TODO: find out how vsmile accesses these GPIO regs!
+	return what;
+}
 
-	if (m_spg243_mode == SPG243_VII)
+
+
+uint16_t spg2xx_game_state::do_spg243_batman_io(uint16_t what, int index)
+{
+	if (index == 0)
 	{
-		if(index == 1)
-		{
-			uint32_t bank = ((what & 0x80) >> 7) | ((what & 0x20) >> 4);
-			switch_bank(bank);
-		}
+		uint16_t temp = m_io_p1->read();
+		what |= (temp & 0x0001) ? 0x8000 : 0;
+		what |= (temp & 0x0002) ? 0x4000 : 0;
+		what |= (temp & 0x0004) ? 0x2000 : 0;
+		what |= (temp & 0x0008) ? 0x1000 : 0;
+		what |= (temp & 0x0010) ? 0x0800 : 0;
+		what |= (temp & 0x0020) ? 0x0400 : 0;
+		what |= (temp & 0x0040) ? 0x0200 : 0;
+		what |= (temp & 0x0080) ? 0x0100 : 0;
 	}
-	else if (m_spg243_mode == SPG243_BATMAN)
+
+	if (index == 2)
```

Comment:
I realise this was here before, but if you have any idea what it's supposed to catch, could you put a `TODO` comment inside the braces to enlighten the rest of us?